### PR TITLE
update arduino to 1.8.9

### PR DESCRIPTION
**Pull request by bergey**
###### Motivation for this change

Update to latest upstream version

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/arduino/arduino-core/default.nix [hunk 1]
```diff
@@ -84,7 +84,7 @@ stdenv.mkDerivation rec {
     url = "http://downloads.arduino.cc/arduino-${version}-${teensy_architecture}.tar.xz";
     sha256 =
       lib.optionalString ("${teensy_architecture}" == "linux64")
-        "1f8s3by5lc6fazyaa9zc9kz3ar8zj8jabab1fy5jzh49fbd8bydx"
+        "1lv4in9j0r8s0cis4zdvbk2637vlj12w69wdxgcxcrwvkcdahkpa"
       + lib.optionalString ("${teensy_architecture}" == "linux32")
         "1r9ral9aq5vp02dwgagifk5h403l7knxdyi1w23rqpcbbpa423lw"
```
Comment (jonringer) [inline pkgs/development/arduino/arduino-core/default.nix]: the other architectures should be updated as well, otherwise they will be broken.

**Comment by bergey** [comment]
`nix-review` caught that my first commit broke `teensyduino`.  Fixed in f95cd76.  `nix-review` now passes.

**Comment by FRidh** [comment]
Please fix your commit messages to conform to our guidelines.

**Review by bjornfor — CHANGES_REQUESTED** [review]
Please reorder commits to not break git bisect.

**Comment by bjornfor** [comment]
I reordered and pushed manually: 51c0c8867448e7d181207510774dac5eb2d9fc91, 98582c6b5236dcd13a1f5bed68ee45dae6e40af8. Thanks!

**Pull request closed without merging**